[IMP] base: make module's Description HTML multilingual

royle-vietnam opened the pull request:
Currently, the description of the module in the Apps menu does not support multiple languages. This PR prioritizes searching for the index_xx.html file based on the language to enable multilingual support.

The description_html field is computed and not stored, so it does not affect stability.




---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/186574.png)](https://mergebot.odoo.com/odoo/odoo/pull/186574)

royle-vietnam commented:
Hi @HydrionBurst, can you review this PR, thanks

HydrionBurst commented:
Technically, it works. But
1. please also fallback to `lang.split('_')[0]` since usually you won't write index.xml for fr_BE and fr_FR and fr_xxx.
2. In 1, there is a tricky case for `es` spanish, see `get_po_paths` in `translate.py`, if we want everything to be consistent.
3. technically, we should mark `ir.module.module.description_html` `depends_context('lang')`

@oomsveta what do you think?
currently the translations for `ir.module.module,description` are all handled in `base`
if we allow index.xml to be logically translatable it might be harder for your team to keep tracking of all translatable content.
maybe your team have other solutions for this problem

royle-vietnam commented:
> Technically, it works. But
> 
> 1. please also fallback to `lang.split('_')[0]` since usually you won't write french index_fr_FR.html index_fr_BE.html, .....
> 2. In 1, there is a tricky case for `es` spanish, see `get_po_paths` in `translate.py`, if we want everything to be consistent.
> 3. technically, we should mark `ir.module.module.description_html` `depends_context('lang')`
> 
> @oomsveta what do you think? currently the translations for `ir.module.module,description` are all handled in `base` if we allow index.xml to be logically translatable it might be harder for your team to keep tracking of all translatable content. maybe your team have other solutions for this problem

Thanks @HydrionBurst  for your suggestion, I have fixed it, please review again.

oomsveta commented on odoo/addons/base/models/ir_module.py:
```diff
@@ -191,8 +191,22 @@ def get_module_info(cls, name):
     def _get_desc(self):
+        def _get_first_valid_index_path(module_path):
+            lang = self.env.lang or 'en_US'
+            lang_base = lang.split('_')[0]
```
what about pt_BR vs pt_PT? sr@latin?

HydrionBurst commented:
One big advantage I can see for the PR is
Currently `ir.module.module.description` (as a fallback for `ir.module.module.description_html` if there is no index.html) can only be translated in the `base` module.
For a third party module, which cannot inject its translations to our `base` module, developers can use this trick to translate module description themselves by using index_xxx.html in their own modules. Their customers only have to download their modules to the correct directory and everything will be translated.
